cmd/publish.go

```diff
@@ -0,0 +1,180 @@
+				handleError(err)
+
+				if !isDescriptionDecorator {
+					continue
+				}
+
+				description, pathArgument, err := checker.GetDecoratorArguments(decorator)
+
+				if len(pathArgument) > 0 {
+					path, ok := pathArgument["path"]
+					if !ok {
+						printErrorAndExit(
+							errors.New("there's no 'path' argument specified inside Description decorator in rule: " +
+								filepath.Join(rulesetPath, rule.Path)))
+					}
+					descriptionBytes, err := os.ReadFile(filepath.Join(rulesetPath, rule.Path, "..", path))
+					handleError(err)
+					currentRuleForRequest.Description = string(descriptionBytes)
+					break
+				} else if len(description) > 0 {
+					currentRuleForRequest.Description = description[0]
+				} else {
+					printErrorAndExit(errors.New("incorrect 'Description' decorator syntax"))
+				}
+				break
+			}
+
+			requestBody.Rules = append(requestBody.Rules, currentRuleForRequest)
+		}
+
+		var tokenFilenamePath string
+
+		tokenFilenamePath = filepath.Join(
+			shared.GetEnvWithDefaultValue("XDG_STATE_HOME", secretGlobalDirectoryDefaultValue),
+			secretDirectoryName,
+			tokenStorageFilename)
+		err = shared.ExpandTilde(&tokenFilenamePath)
+		handleError(err)
+
+		isTokenStoredLocally, err := cmd.Flags().GetBool("local")
+		handleError(err)
+
+		tokenFileRawData, err := os.ReadFile(tokenFilenamePath)
+		handleError(err)
+
+		tokenData := TokenStorageLayout{}
+		err = bson.Unmarshal(tokenFileRawData, &tokenData)
+		handleError(err)
+
+		var token string
+		if isTokenStoredLocally {
+			tokenPosition := slices.IndexFunc(tokenData.LocalKeys, func(localToken LocalToken) bool {
+				return localToken.Path == rulesetPath
+			})
+			if tokenPosition == -1 {
+				printErrorAndExit(errors.New("cannot find the local token for ruleset: " + rulesetPath))
+			}
+			token = tokenData.LocalKeys[tokenPosition].Token
+		} else {
+			token = tokenData.GlobalToken
+		}
+
+		jsonBody, err := json.Marshal(requestBody)
+		handleError(err)
+
+		requestUrl, err := url.JoinPath(os.Getenv("BACKEND_URL"), publishRulesetRoute)
+		handleError(err)
+
+		httpRequest, err := http.NewRequest("POST", requestUrl, bytes.NewBuffer(jsonBody))
+		handleError(err)
+
+		httpRequest.Header.Add("Authorization", "Token "+string(token))
+		httpRequest.Header.Add("Content-Type", "application/json")
+
+		httpClient := http.Client{}
+
+		httpResponse, err := httpClient.Do(httpRequest)
+		handleError(err)
+		defer func() {
+			err = httpResponse.Body.Close()
+			handleError(err)
+		}()
+
+		var responseBody map[string]interface{}
+		err = json.NewDecoder(httpResponse.Body).Decode(&responseBody)
+
+		if statusCode, codeExists := responseBody["statusCode"]; codeExists && statusCode.(float64) == 404 {
+			printErrorAndExit(errors.New("it seems that your ruleset doesn't exist in the spito store. Please create it in the spito GUI first"))
+		} else if codeExists && statusCode.(float64) != 200 {
```

Comment:
- Why are you comparing float64 type to integer
- Instead check if code starts with `2`. For example status code `201` shouldn't be considered as an error